apps/server/src/modules/files-storage/service/files-storage.service.spec.ts

```diff
@@ -0,0 +1,814 @@
+				const { params, fileRecords } = setup();
+
+				const result = await service.getFilesOfParent(params);
+
+				expect(result).toEqual([fileRecords, 3]);
+			});
+		});
+
+		describe('WHEN repository throws an error', () => {
+			const setup = () => {
+				const { params } = getFileRecordsWithParams();
+
+				fileRecordRepo.findBySchoolIdAndParentId.mockRejectedValueOnce(new Error('bla'));
+
+				return { params };
+			};
+
+			it('should pass the error', async () => {
+				const { params } = setup();
+
+				await expect(service.getFilesOfParent(params)).rejects.toThrow(new Error('bla'));
+			});
+		});
+	});
+
+	describe('patchFilename is called', () => {
+		describe('WHEN file(s) exists and valid params are passed', () => {
+			let spy: jest.SpyInstance;
+
+			afterEach(() => {
+				spy.mockRestore();
+			});
+
+			const setup = () => {
+				const { fileRecords, params } = getFileRecordsWithParams();
+				const fileRecord = fileRecords[0];
+				const data: RenameFileParams = { fileName: 'renamed' };
+
+				spy = jest.spyOn(service, 'getFilesOfParent').mockResolvedValueOnce([fileRecords, 1]);
+
+				return {
+					data,
+					fileRecord,
+					fileRecords,
+					params,
+				};
+			};
+
+			it('should call getFilesOfParent with right paramaters', async () => {
+				const { fileRecord, data } = setup();
+				const fileRecordParams = filesStorageHelper.mapFileRecordToFileRecordParams(fileRecord);
+
+				await service.patchFilename(fileRecord, data);
+
+				expect(spy).toHaveBeenCalledWith(fileRecordParams);
+			});
+
+			it('should call fileRecordRepo.save with right paramaters', async () => {
+				const { fileRecord, data } = setup();
+				const expectedFileRecord = _.cloneDeep(fileRecord);
+				expectedFileRecord.name = data.fileName;
+
+				await service.patchFilename(fileRecord, data);
+
+				expect(fileRecordRepo.save).toHaveBeenCalledWith(expectedFileRecord);
+			});
+
+			it('should return modified fileRecord', async () => {
+				const { fileRecord, data } = setup();
+
+				const result = await service.patchFilename(fileRecord, data);
+
+				expect(result.name).toEqual(data.fileName);
+			});
+		});
+
+		describe('WHEN repository is throwing an error', () => {
+			const setup = () => {
+				const { fileRecord, params } = getFileRecordWithParams();
+				const data: RenameFileParams = { fileName: 'renamed' };
+
+				const spyGetFilesOfParent = jest.spyOn(service, 'getFilesOfParent').mockResolvedValueOnce([[fileRecord], 1]);
+				fileRecordRepo.save.mockRejectedValueOnce(new Error('bla'));
+
+				return {
+					data,
+					fileRecord,
+					params,
+					spyGetFilesOfParent,
+				};
+			};
+
+			it('should pass the error', async () => {
+				const { fileRecord, data } = setup();
+
+				await expect(service.patchFilename(fileRecord, data)).rejects.toThrowError(new Error('bla'));
+			});
+		});
+	});
+
+	describe('updateSecurityStatus is called', () => {
+		describe('WHEN file exists and no virus is found', () => {
```

Comment:
Das Verhalten vom Helper testen wir schon im Unittest des Helpers. Deswegen bräuchte es hier eigentlich nicht zwei describe Blöcke für Virus gefunden oder nicht.

Reply:
Daher "WHEN file exists and a virus is found" raus werfen und den hier umformulieren?

Die Frage wäre hier, ist es part des service zu spefizieren wie im Falle eines block status gehandelt werden soll, oder nicht.

Reply:
Ja genau, das dachte ich. Ich würde es nur im Helper testen. Da verlassen wir uns an anderen Stelle ja auch drauf. Warum sollte es hier anders sein?

Reply:
Diskussion nicht final geklärt. Tests bleiben zunächst drin.